sim/paladin/righteous_vengeance.go

```diff
@@ -0,0 +1,75 @@
+package paladin
+
+import (
+	"time"
+
+	"github.com/wowsims/sod/sim/core"
+	"github.com/wowsims/sod/sim/core/proto"
+)
+
+const RVTicks = 4
+
+func (paladin *Paladin) registerRV() {
+	if !paladin.hasRune(proto.PaladinRune_RuneCloakRighteousVengeance) {
+		return
+	}
+
+	paladin.RegisterAura(core.Aura{
+		Label:    "Righteous Vengeance",
+		Duration: core.NeverExpires,
+		OnReset: func(aura *core.Aura, sim *core.Simulation) {
+			aura.Activate(sim)
+		},
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+
+			if spell.Flags.Matches(SpellFlag_RV) && result.DidCrit() {
+				paladin.procRV(sim, result)
+			}
+		},
+	})
+
+	paladin.rv = paladin.RegisterSpell(core.SpellConfig{
+		ActionID:    core.ActionID{SpellID: 440675},
+		SpellSchool: core.SpellSchoolHoly,
+		DefenseType: core.DefenseTypeMelee,
```

Comment:
I could be wrong but I think this should maybe be `DefenseTypeMagic`. That will cause it to scale with magic / holy damage taken effects whereas right now I think this might scale with physical damage taken debuffs

Reply:
![image](https://github.com/wowsims/sod/assets/5923944/d3a6714d-1615-4d9e-8c7c-4a0ac9df19a4)